backend/src/services/auth/auth-fns.ts

```diff
@@ -44,3 +44,24 @@ export const validateSignUpAuthorization = (token: string, userId: string, valid
   if (decodedToken.authTokenType !== AuthTokenType.SIGNUP_TOKEN) throw new UnauthorizedError();
   if (decodedToken.userId !== userId) throw new UnauthorizedError();
 };
+
+export const enforceUserLockStatus = (isLocked: boolean, temporaryLockDateEnd?: Date | null) => {
```

Comment:
What if the email is not send or for some reason failed. Is there a way for user to re-send it again?

Reply:
none. In this case (which currently is configured to happen only after the 12th failed MFA attempt), users will have to contact support or their administrators if they don't receive any email

Reply:
Discussed this with @maidul98, we'll be allowing account unlock via password reset as well 😄 
<img width="1504" alt="image" src="https://github.com/Infisical/infisical/assets/65645666/5714331e-a77f-4c0f-98e9-af60be1cf000">

cc: @dangtony98